common/src/main/java/io/druid/metadata/DefaultPasswordProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2015 - Yahoo! Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.metadata;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.ISE;
+
+
+public class DefaultPasswordProvider implements PasswordProvider
+{
+
+  private final static String PASSWORD_KEY = "password";
+
+  private final String password;
+
+  @JsonCreator
+  public DefaultPasswordProvider(Object spec)
+  {
+    password = convertValue(spec);
+  }
+
+  private static String convertValue(Object spec) {
+    if(spec instanceof String) {
+      return spec.toString();
+    } else if(spec instanceof Map && ((Map)spec).containsKey(PASSWORD_KEY)) {
```

Comment:
ah ok, I understood the need for string, but I was confused about the need for Map.

Since you already declare a `default` type as part of your jackson annotation, I don't see why you need to parse the map directly. You should be able to just add a `@JsonProperty("password") password` field to you constructor.

You can then add a static fromString method to a separate class and make that the `defaultImpl` for jackson.

Have a look at [QueryGranularity](https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/granularity/QueryGranularity.java#L46) where we do something similar, with a fromString method annotated with `@JsonCreator` that instantiates the default, but everything else using properly annotated constructors.